spyder/widgets/projects/tests/test_project_explorer.py

```diff
@@ -15,12 +19,38 @@
 from spyder.widgets.projects.explorer import ProjectExplorerTest
 
 @pytest.fixture
-def setup_projects_explorer(qtbot):
+def setup_projects_explorer(qtbot, directory=None):
     """Set up ProjectExplorerWidgetTest."""
-    project_explorer = ProjectExplorerTest()
+    project_explorer = ProjectExplorerTest(directory=directory)
     qtbot.addWidget(project_explorer)
     return project_explorer
 
+
+def test_change_directory_in_project_explorer(qtbot, tmpdir):
```

Comment:
You need to pass the fixture as an argument to the test, else it can't be initialized correctly.